encoder/libav_encoder.cpp

```diff
@@ -192,8 +191,17 @@ void LibAvEncoder::initVideoCodec(VideoOptions const *options, StreamInfo const
 	}
 	else
 		format = options->libav_format.c_str();
+
+	// Legacy handling of the --listen parameter.  If missing from the url string, add "?listen=1" to the end.
+	if (options->listen && output_file_.find(tcp.c_str(), 0, tcp.length()) != std::string::npos)
+	{
+		const std::string listen { "?listen=1" };
+		if (output_file_.find(listen, output_file_.length() - listen.length()) == std::string::npos)
```

Comment:
Just slightly wondering whether it's necessary to constrain the length of the search here, wouldn't that happen by itself?

Reply:
True, I can get rid of the length constraint.

Reply:
Actually, this is the equivalent of ends_with(), so the constraint here is the start position of the search.